### DISCUSSION TITLE
Error HH411: The library src, imported from @eth-optimism/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol, is not installed. Try installing it using npm.

### DISCUSSION THREAD

**Discussion by Chomtana** [NomicFoundation/hardhat#5049] [Troubleshooting]
@eth-optimism/contracts-bedrock is using foundry with src/ import but I need to import it to a hardhat project. How should I do this?

**Comment by kanej** [comment]
To clarify my understanding here, you have installed `@eth-optimism/contracts-bedrock` via npm and imported `@eth-optimism/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol` into a contract under `./contracts` of a Hardhat project.

Compilation then fails as, internally the `@eth-optimism/contracts-bedrock` contracts import statements don't include the `/src` part of the file path (as they are doing that with a remapping on the Foundry side). Hardhat doesn't know about the remappings so errors on the import statements in Solidity files? Is that what you are seeing?

**Reply by Chomtana** [reply]
Yes, I have installed `@eth-optimism/contracts-bedrock` via npm and imported `@eth-optimism/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol` into a contract under `./contracts` of a Hardhat project.

You can look into it at: https://github.com/Opti-domains/evmgateway/tree/op-fault-proof/op-verifier

And `IDisputeGame.sol` is available at https://github.com/ethereum-optimism/optimism/blob/v1.7.2/packages/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol

Currently, I extract IDisputeGame.sol into an interface under `./contracts` for a temporary fix to this problem. But I need to import `@eth-optimism/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol` in order to merge into the upstream repo.

**Reply by Chomtana** [reply]
Any solution?

**Reply by kanej** [reply]
I think this is an issue with the way Hardhat and Foundry resolve Solidity imports and they clash - Foundry resolution is based on remappings, while Hardhat has a treats direct imports as coming from the `node_modules` folder.

We are looking to add first class support for remappings to Hardhat to support these sorts of cases.

**Reply by lakshay-zeeve** [reply]
Any resolution to this problem?

**Reply by kanej** [reply]
We have added remapping support to the Hardhat 3 Alpha: https://hardhat.org/hardhat3-alpha
We are working on adding better docs to cover the new capabilities.

**Comment by amit-shah-cb** [comment]
Same issue here

**Reply by lakshay-zeeve** [reply]
Any resolution to this problem?
